dao/src/main/java/com/epam/esm/dao/specification/tag/GetAllTagsAssociatedWithGiftCertificates.java

```diff
@@ -0,0 +1,31 @@
+package com.epam.esm.dao.specification.tag;
+
+import com.epam.esm.dao.entity.GiftCertificate;
+import com.epam.esm.dao.entity.Tag;
+import com.epam.esm.dao.specification.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Root;
+
+public final class GetAllTagsAssociatedWithGiftCertificates implements Specification<Tag> {
```

Comment:
Rename it please. Is it GetTagsByCertificateSpecification?

Reply:
Thank you. You're absolutely right. (fixed)